python-threatexchange/threatexchange/signal_type/pdq_index.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+Implementation of SignalTypeIndex abstraction for PDQ by wrapping hashing.pdq_faiss_matcher.
+"""
+
+import typing as t
+import pickle
+
+from threatexchange.signal_type.index import SignalTypeIndex, IndexMatch
+from threatexchange.hashing.pdq_faiss_matcher import PDQMultiHashIndex
+
+T = t.TypeVar("T")
+PDQ_CONFIDENT_MATCH_THRESHOLD = 31
+
+
+class PDQIndex(SignalTypeIndex):
+    """
+    Wrapper around the pdq faiss index lib
+    """
+
+    def __init__(self, entries: t.Iterable[t.Tuple[str, T]]) -> None:
+        super().__init__()
+        self.local_id_to_entry = dict()
+        hashes = list()
+        for i in range(len(entries)):
```

Comment:
blocking: This is actually not safe, since Iterable is a generator, it won't have a length.

Instead do

```
for i, entry in enumerate(entries):
```